### ISSUE TITLE
setWith extraArgs

### ISSUE THREAD

**Issue by alexesDev** [planttheidea/unchanged#37]
Hi. Why setWith not working?
```js
setWith((_, x) => x, 'test', { test: 1 }, 2);
{ test: [Function: bound slice] }
```
Also
```js
const fn: unchanged.withHandler = (value: any, preventUpdate: boolean): any =>
  preventUpdate ? currentValue : "quz";

console.log(setWith(fn, "foo[0].bar", object)); // {foo: [{bar: 'quz'}]}
```
Print
```js
{ foo: [ { bar: 'baz' } ] }
```

**Comment by alexesDev** [comment]
This is a very strange code :open_mouth: 
```js
const slice = Function.prototype.bind.call(Function.prototype.bind, Array.prototype.slice);
```
Why not just `Array.prototype.slice.call`?

**Comment by planttheidea** [comment]
Ha okay so this is a stupid issue ... the first param of that setup should be `Function.prototype.call`, not `Function.prototype.bind`. Not sure why my testing didn't catch this, but I'll update.

To answer your question, it avoids property access. It's a micro-optimization that had a slight perf win in specific legacy environments.

**Comment by planttheidea** [comment]
This should be fixed in `2.2.1`. Thanks for calling this out!
